src/controllers/tripRequestController.js

```diff
@@ -0,0 +1,205 @@
+      },
+    })
+      .then((data) => {
+        data !== null
+          ? res.json({ response: data })
+          : res.status(404).json({ response: 'No trip found' });
+      })
+      .catch((error) => {
+        res.json({ response: error.parent.detail });
+      });
+  };
+
+  static viewSingleTripByManager = async (req, res) => {
+    const { user } = req;
+    const reqId = user.user.id;
+    const { id } = req.params;
+    // const numberValidator = await isNumber(id);
+    // if (numberValidator !== true) {
+    //   return res.status(400).json({ message: 'id must be a number' });
+    // }
+    await TripRequest.findOne({
+      where: {
+        managerId: reqId,
+        id,
+      },
+    })
+      .then((data) => {
+        data !== null
+          ? res.json({ response: data })
+          : res.status(404).json({ response: 'No trip found' });
+      })
+      .catch((error) => {
+        res.json({ response: error.parent.detail });
+      });
+  };
+
+  static viewTripsByManager = async (req, res) => {
+    const { user } = req;
+    await TripRequest.findAll({
+      where: {
+        managerId: user.user.id,
+      },
+    })
+      .then((data) => {
+        res.json({ response: data });
+      })
+      .catch((error) => {
+        res.json({ response: error.parent.detail });
+      });
+  };
+
+  static deleteTripRequest = async (req, res) => {
+    const { user } = req;
+    const reqId = user.user.id;
+    const { id } = req.params;
+    // const numberValidator = await isNumber(id);
+    // if (numberValidator !== true) {
+    //   return res.status(400).json({ message: 'id must be a number' });
+    // }
+    await TripRequest.destroy({
+      where: {
+        requesterId: reqId,
+        status: 'pending',
+        id,
+      },
+    })
+      .then((data) => {
+        // eslint-disable-next-line no-unused-expressions
+        data > 0
+          ? res.json({ response: 'Trip request deleted successfully' })
+          : res.status(404).json({ response: 'Trip request not found' });
+      })
+      .catch((error) => {
+        res.json({ response: error });
+      });
+  };
+
+  static updateTripRequest = async (req, res) => {
+    const { user } = req;
+    const reqId = user.user.id;
+    const { id } = req.params;
+    // const numberValidator = await isNumber(id);
+    // if (numberValidator !== true) {
+    //   return res.status(400).json({ message: 'id must be a number' });
+    // }
+    await TripRequest.update(req.body, {
+      where: {
+        requesterId: reqId,
+        status: 'pending',
+        id,
+      },
+    })
+      .then((data) => {
+        // eslint-disable-next-line no-unused-expressions
+        data > 0
+          ? res.json({ response: 'Trip request updated successfully' })
+          : res.status(404).json({ response: 'Trip request not found' });
+      })
+      .catch((error) => {
+        res.json({
+          response: 'Ooops, something went wrong. trip request is not created',
```

Comment:
@Robertishimwe  Improve the validation error messages for example ``Ooops trip request can't be created, something went wrong. ``